src/dune_pkg/fetch.ml

```diff
@@ -260,6 +261,123 @@ let fetch ~unpack ~checksum ~target (url : OpamUrl.t) =
         url)
 ;;
 
+module Git_repo = struct
+  type t =
+    { git : Path.t
+    ; dir : Path.t
+    }
+
+  type rev = Rev of string
+
+  let equal { git; dir } t = Path.equal git t.git && Path.equal dir t.dir
+  let display = Display.Quiet
+  let failure_mode = Process.Failure_mode.Strict
+  let run { git; dir } = Process.run ~dir ~display failure_mode git
```

Comment:
Yep, I just didn't want to deal with it while I was getting out the kinks. Maybe we should really have a new display mode where the commands are silent? @rgrinberg what do you think?

Reply:
I think that would necessarily have to be user facing and I am not sure the utility in having no output from Dune. I definitely wouldn't encourage it, but it could be useful for other programs wishing to run dune.

Reply:
I don't mean making dune silent, I mean adding a display mode where processes don't write to the users terminal. There is no need for the user to even know `git` is called in the background do create a bare repo on demand.

Reply:
> Maybe we should really have a new display mode where the commands are silent?

The display mode is something that corresponds to `--display`, so I'm sure if that's the right place. We do have flags like `--action-stdout-on-success`. Maybe they should be the default.

> I mean adding a display mode where processes don't write to the users terminal.

Isn't that what you already did with `make_stdout` and `make_stderr`?

> There is no need for the user to even know git is called in the background do create a bare repo on demand.

If the operation requires network access, I think it's useful to tell the user what is running. I agree that other commands don't need to be displayed.

In any case, the PR doesn't need to fix these issues before it's good to go.